src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -769,12 +962,47 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
                 binder,
                 conversions,
                 diagnostics,
+                useConstructorExitWarnings,
                 useDelegateInvokeParameterTypes: false,
                 delegateInvokeMethodOpt: null,
-                initialState: null,
+                initialState: initialNullableState,
                 analyzedNullabilityMapOpt: null,
                 snapshotBuilderOpt: null,
-                returnTypesOpt: null);
+                returnTypesOpt: null,
+                getFinalNullableState,
+                finalNullableState: out finalNullableState);
+        }
+
+        /// <summary>
+        /// Gets the "after initializers state" which should be used at the beginning of nullable analysis
+        /// of certain constructors. Only used for semantic model and debug verification.
+        /// </summary>
+        internal static VariableState? GetAfterInitializersState(CSharpCompilation compilation, Symbol? symbol)
+        {
+            if (symbol is MethodSymbol method
+                && method.IncludeFieldInitializersInBody()
+                && method.ContainingType is SourceMemberContainerTypeSymbol containingType)
+            {
+                var unusedDiagnostics = DiagnosticBag.GetInstance();
+
+                Binder.ProcessedFieldInitializers initializers = default;
+                Binder.BindFieldInitializers(compilation, null, method.IsStatic ? containingType.StaticInitializers : containingType.InstanceInitializers, unusedDiagnostics, ref initializers);
```

Comment:
>Binder.BindFieldInitializers [](start = 16, length = 28)

Are we binding the initializers a second time, in addition to initial binding?

Reply:
In SemanticModel use cases this may be the first time we bind the initializers, but we will have to bind initializers again to get all diagnostics from the semantic model.

After chat with @333fred we decided that this was a reasonable first resort. We could also consider caching the bound initializers or the `AfterInitializersState` for a set of classes on a semantic model if we found there was a significant performance drag from doing it this way.